docs/DesignDocs/SealingAPI.md

```diff
@@ -0,0 +1,591 @@
+ * behalf of the caller.
+ * @param[in] entropy_size Must be \c 0 if \p entropy is \c NULL.
+ * @param[in] tee_specifc \c OE_SEAL_TEE_AGNOSTIC to use TEE defaults. If \c
+ * OE_SEAL_SGX is specified, the rest of \p tee_specifc is interpreted as the
+ * \a attribute_mask for \c EGETKEY on SGX.
+ * @param[out] key_info On success this points to the enclave-specific key
+ * information which can be used to retrieve the key by passing it to
+ * oe_get_seal_key_v2(). Freed by calling oe_free_seal_key() or oe_free_key().
+ * @param[out] key_info_size On success, this is the size of the \p key_info
+ * buffer.
+ *
+ * @retval OE_OK The seal key info was successfully requested.
+ * @retval OE_INVALID_PARAMETER At least one parameter is invalid.
+ * @retval OE_OUT_OF_MEMORY Failed to allocate memory.
+ */
+oe_result_t oe_get_seal_key_info(
+    oe_seal_policy_t seal_policy,
+    uint8_t* entropy,
+    size_t entropy_size,
+    uint64_t tee_specifc,
+    uint8_t** key_info,
+    size_t* key_info_size);
+```
+
+`oe_get_seal_key_info()` duplicates largely the logic of
+`oe_get_seal_key_by_policy()`. Please see [User Experience](#user-experience)
+earlier in this documentation for an explanation on the intention of this
+API.
+
+`seal_policy` selects the type of enclave identity and has the same definition
+and supported values as in `oe_get_seal_key_by_policy()`.
+
+`entropy`/`entropy_size` supplies additional data as input to the KDF.
+Different TEEs may expect different entropy sizes, e.g., SGX expects 256-bit
+entropy. If `entropy_size` isn't the same as expected, `entropy` will be hashed
+before being passed on to the underlying TEE. `entropy` could be `NULL` and if
+that's the case, `entropy_size` must be `0`, and a random bit sequence of
+expected size will be used instead.
+
+`tee_specific` is a 64-bit TEE specific parameter. Its interpretation varies
+from TEE to TEE. It could be an integer on one TEE, or a pointer to some
+structure on another. Its purpose is to pass additional parameters that need to
+be tuned frequently, e.g., `attribute_mask` on SGX. It doesn't intend to be
+comprehensive, as fine tunes can always be done by casting `key_info` into the
+TEE specific structure (e.g., `sgx_key_request_t` on SGX) and modifying it
+there.
+
+Values of `tee_specific` currently defined are listed in the table below.
+
+|Value|TEE|Definition|
+|-----|---|----------|
+|`OE_SEAL_TEE_AGNOSTIC`|*All*|No TEE specific features are requested. Signify to use TEE defaults.
+|`OE_SEAL_SGX \| <attribute_mask>`|**SGX**|`<attribute_mask>` specifies `sgx_key_request_t::attribute_mask` as input to `EGETKEY`.
+
+`OE_SEAL_TEE_AGNOSTIC` is TEE neutral. Defining it to zero should suffice.
+
+```C
+#define OE_SEAL_TEE_AGNOSTIC 0
+```
+
+`OE_SEAL_SGX` should be defined **if and only if** SGX is the target TEE, in
+order to guarantee a compiler error should the code be compiled targeting a
+different TEE. The value `SGX_FLAGS_INITED` is recommended as it can never be
```

Comment:
Is SGX_FLAGS_INITED a literal define from an Intel header file?   (It doesn't start with OE_ hence my question, since if defined in an OE header file it should be OE_SGX_FLAGS_INITIALIZED or similar.)

Reply:
`SGX_FLAGS_INITED` is an existing macro defined in `bits/sgx/sgxtypes.h`. Shall I submit another pull request to change it?

Reply:
That would be great if you could, thanks!